New website UI

AdrianoCahete opened the pull request:
Fixes #92.

AdrianoCahete committed:
Add uptodate django links

AdrianoCahete committed:
Remove unused files

AdrianoCahete committed:
Add base css file

AdrianoCahete committed:
Fix base template code to match layout proposal

AdrianoCahete committed:
Add logo proposal assets

AdrianoCahete committed:
Add new footer and nav starter styles + Add OpenSans typography

AdrianoCahete committed:
Removing map file to keep logs clean

AdrianoCahete committed:
Move brand images to brand folder

AdrianoCahete committed:
Remove typography for now

AdrianoCahete committed:
Add new preview brand logo

AdrianoCahete committed:
Add new base layout for footer and navbar

AdrianoCahete committed:
New specs for menu

AdrianoCahete committed:
Modal splitted style

AdrianoCahete committed:
Move content to match actual layout

AdrianoCahete committed:
Fix footer order

AdrianoCahete committed:
Better header/nav + Better Hero block + Better footer

AdrianoCahete committed:
Fix index content order

AdrianoCahete committed:
Add button, hero, content and some utilities classes + Fix modal position

AdrianoCahete committed:
Add TODO to download component to see later + Remade sponsor component to match new UI + Move main element from content to main to match W3C actual specs + Move components on index to better match new UI design

AdrianoCahete committed:
Fix button styles + Add myself as a designer on About page + Fix navigation alignment

AdrianoCahete commented:
By now it's usable, but not so great if anyone wants to see in a live environment. My plan is to finish, at least the first usable version att he end of the week,

AdrianoCahete committed:
Add github link to the footer

williamdes commented on pmaweb/static/images/brand/phpMyAdmin-logo-preview.svg:
```diff
@@ -0,0 +1,6 @@
+<svg fill="none" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1020 792">
```
This beautiful icon would be accepted into our data repository. You can open a pull-request to add new assets anytime :)

AdrianoCahete replied:
I have a friend that is doing a fine tune on that logo, as soon she finishes, I'll do the PR with the brand guide.

AdrianoCahete committed:
Fix some Lighthouse warnings + Fix footer layout +Move svg icons to a component to avoid cluttered base html

AdrianoCahete committed:
Move component templates to a component folder

AdrianoCahete committed:
Move component templates to a component folder

AdrianoCahete committed:
Move component templates to a component folder

AdrianoCahete committed:
Fix button layout states + Fix order in svg icons to be a better one (pma logos before) + First Hero block version

AdrianoCahete committed:
Fix nav active styles + Fix button styles in modal + Fix default button element to match actual button styles

AdrianoCahete committed:
Add News layout

AdrianoCahete committed:
Remove test content

AdrianoCahete committed:
Add layout to Features block

AdrianoCahete committed:
Fix header styles + Add svg illustrations

AdrianoCahete committed:
Preparing to create the mobile version

AdrianoCahete committed:
Fiux sponsor block style

AdrianoCahete committed:
Fix code style inside downlaod popup + Fix link hover on nav + Remove bootstrap + Fix typo on modal file

AdrianoCahete commented:
Home is finished on desktop. I'm moving to internal pages to make them usable on the new UI and then move to mobile.

Feel free to give me feedbacks :)

AdrianoCahete committed:
Base style fixes for internal pages

AdrianoCahete committed:
Fix table styles + Fix Docs screens

AdrianoCahete committed:
Fix sponsor list style

AdrianoCahete committed:
Fix internal pages + Add some of them on docs page

AdrianoCahete committed:
Fix internal pages + Reorder links on doc page to make more sense

AdrianoCahete committed:
Fix more layouts +Add Security annouces to support and download page + Add link to latest file on download page +Move header download button to download page instead of download file itself directly +

AdrianoCahete committed:
Add and remove TODOs + Remove unused code

AdrianoCahete committed:
Add new sponsor

AdrianoCahete committed:
First accordion style

AdrianoCahete committed:
Accordion styles

AdrianoCahete committed:
Standardize code + Remove unused code

AdrianoCahete committed:
Fix for nav-pills style + Fix for dllist + Fix for theme page

AdrianoCahete committed:
Fixing award page

AdrianoCahete committed:
Fix Team page photos and team's description

AdrianoCahete committed:
Fix Translation page + Fix progress bar styles

AdrianoCahete committed:
Fix Try page

AdrianoCahete committed:
Fix Try page

AdrianoCahete committed:
Remove unused code

AdrianoCahete commented:
I have just a few fixes to do, but I think that it's easier to make this PR for a desktop version and, after merged, go to mobile.

Since the mobile is wrong in some places on the current UI (menu, code block, some tables) and no one opened an issue, I think that most people don't really see the website on mobile... ¯\\_(ツ)_/¯

AdrianoCahete committed:
Mobile base

AdrianoCahete committed:
Mobile fixes + Docs updates + Demo alerts

AdrianoCahete committed:
Update Theme Page

AdrianoCahete committed:
Add Team link to About Page

williamdes commented:
Would you mind posting screenshots please ?

AdrianoCahete commented:
I still need to fix Books (inside Docs), Donate, and History layouts (they are working, but not with the best visual).

A "small" changelog before:
 - Docs URLs to Django 1.7 are removed from the Django site. I found the readthedocs mirrors and changed when I found them.
  - Changed the Download link on the header to move to the Download page since we don't have a link to that page. CTA button on page stills points to download directly.
  - Try button points to the Try page, so the user can see all the warnings before starts. Inside the page, there's a button to Demo env.
  - Added a Download button on Docs page for people who visit with mobile, since the Header CTA buttons will not appear on small resolutions.
  - Removed bootstrap since I don't think that you use neither 30% of the package. I made custom css instead that matches your current use (and may match your future needs). I may write internal documentation for those styles later.
  - Removed FontAwesome too, since all the icons now are SVG's.
  - Move from <content> to <main> element to match new specs (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/content)
  - Made some styles specific so you don't need to read documentation all the time, doesn't break the style guide unintentionally neither use classes where is not needed. (you just can use the `.footer-link` class inside the footer element. You can just use `.btn-big` inside `.btn` element. You will never see a `footer-link` on navbar without some strangeness -- easier to spot on future PR)
  - Moved components to a component folder to better differentiate component/include from templates files
  - I made an alert component, just use `alert` in conjunction with some alert definition (`alert-info`, `alert-error`, `alert-success`, `alert-warning`). You can see it on the Download popup.  
  - Fixed the modal style
  - Fixed the progress bar styles
  - Add the Logo history text
  - Removed unused (and fully deprecated) Google Code from 404 page
  
 Problems that I faced on local env:
  - Cannot run News and Security page, so I made changes directly on html and didn't test on live code itself, only in generated HTML (maaaaaybe because I'm using Windows?)
  - We need an `if` on line 134 (index.html) to not show the year if it's the current. I don't know how to do this on Django.
  - I may have broken the Announcement list accordion on the Security page (Support > Security) (need to view on a more stable env)
  - Download popup here just flashes, but I think that it's because I don't have anything to download. Needs a proper env to test.
 - Some screenshots doesn't have data because I don't have that data on my local env. I don't know why.

---

![Screen Shot 2020-11-07 at 17 05 35-fullpage](https://user-images.githubusercontent.com/37474/98447341-eb85d580-2102-11eb-934d-a2b66714dba7.png)

---

![image](https://user-images.githubusercontent.com/37474/98447529-36541d00-2104-11eb-888b-4f44539c70d2.png)


---

![Screen Shot 2020-11-07 at 17 05 59-fullpage](https://user-images.githubusercontent.com/37474/98447359-0d7f5800-2103-11eb-8d14-24137cd0e51c.png)

---

![Screen Shot 2020-11-07 at 17 06 03-fullpage](https://user-images.githubusercontent.com/37474/98447363-1839ed00-2103-11eb-964e-0efc7abf439c.png)

---

![Screen Shot 2020-11-07 at 17 06 29-fullpage](https://user-images.githubusercontent.com/37474/98447365-1d973780-2103-11eb-8ccb-3b8cbe416dc1.png)

---

![Screen Shot 2020-11-07 at 17 06 52-fullpage](https://user-images.githubusercontent.com/37474/98447370-225beb80-2103-11eb-9fa2-101c1bc64230.png)

---

![Screen Shot 2020-11-07 at 17 07 09-fullpage](https://user-images.githubusercontent.com/37474/98447372-2851cc80-2103-11eb-9589-1f80e03de924.png)

---

![Screen Shot 2020-11-07 at 17 08 16-fullpage](https://user-images.githubusercontent.com/37474/98447374-2c7dea00-2103-11eb-8485-438cb7fc4eaf.png)

---

![Screen Shot 2020-11-07 at 17 08 23-fullpage](https://user-images.githubusercontent.com/37474/98447380-30117100-2103-11eb-8805-7b89ac7db77f.png)

---

![Screen Shot 2020-11-07 at 17 08 29-fullpage](https://user-images.githubusercontent.com/37474/98447384-369fe880-2103-11eb-9130-2d351523f97b.png)

---

![Screen Shot 2020-11-07 at 17 09 40-fullpage](https://user-images.githubusercontent.com/37474/98447389-3acc0600-2103-11eb-85d7-5d15efe96db2.png)

AdrianoCahete commented:
On Mobile works, but it's pretty broken right now (gray block is the menu that I didn't put the icon yet, so made gray just to find the click area).

![Screen Shot 2020-11-07 at 17 22 46-fullpage](https://user-images.githubusercontent.com/37474/98447593-c4c89e80-2104-11eb-8b5f-a059c9ff19fc.png)

williamdes commented:
Very good work !

I will print this logo on a t-shirt ASAP when it is uploaded to the data repo :heart_eyes:

AdrianoCahete committed:
Remove unused code

AdrianoCahete committed:
Fix Donate, Docs and About page layouts + Add History view to About page + Fix css variables + Prepare for future dark mode

AdrianoCahete committed:
Fix typos + Fix menu on mobile + Fix footer on small mobile

AdrianoCahete commented:
Fixed the mobile now just... why not?

![Screen Shot 2020-11-10 at 04 18 25-fullpage](https://user-images.githubusercontent.com/37474/98627295-aaecae80-22f2-11eb-8965-fe2b4820d99e.png)

AdrianoCahete committed:
Update favicons with preview v2

AdrianoCahete commented:
I think that we can proceed with review and fixes, then wait for the new brand to become available (I hope that the end of this week) to update it and merge.

We already finished the logo as a symbol itself, we're trying a few OSS typographies to match a better visual.

Conflicting files are because I changed the path and the elements inside, but the content is the same.

AdrianoCahete committed:
Dark Theme Preview :)

AdrianoCahete commented:
Doing a lot of nothing these months, I made a preview Dark Theme (that respects browser definitions!) for the website, but since it's heavily untested, I prefer to leave commented out. if you want to proceed with that in the future, it's already in the code :)

![127 0 0 1_8000_](https://user-images.githubusercontent.com/37474/98632287-02444c00-22fe-11eb-8a23-d4f1cf087652.png)

AdrianoCahete commented:
Menu @ Mobile (to be honest, every screen below than 900px wide)

![menu](https://user-images.githubusercontent.com/37474/98632862-163c7d80-22ff-11eb-95b9-4b23c3c4de96.gif)

williamdes reviewed (approved):
LGTM
mobile previews looks great !

AdrianoCahete commented:
Default Logo
<img width="242" alt="phpMyAdmin-Logo-Default" src="https://user-images.githubusercontent.com/37474/98715890-82ec6200-2369-11eb-846b-65752f169ca1.png">

Symbol (when there's no space to use the default logo, like favicons)
<img width="96" alt="phpMyAdmin-Logo-Symbol" src="https://user-images.githubusercontent.com/37474/98715893-8384f880-2369-11eb-995d-f3c1bbeca22b.png">

There's another one to use in dark backgrounds, but is only the symbol with a circle behind it. We don't have the full logo for use in dark backgrounds yet, but I don't think that it's the main concern right now.

The typography used was [Montserrat](https://fonts.google.com/specimen/Montserrat#about), but I'll make the update with all the infos in the Brand Guide that we'll be made availiable later.

I'll update that PR and the data repository with the new logos by the end of the day.

AdrianoCahete committed:
Update Brand assets

AdrianoCahete commented:
Ok, all brand is updated. Just waiting for you to test and merge! :)

AdrianoCahete commented:
JFYI: I don't how do you handle the conflicts and merge strategies, but I don't have access to either.

williamdes commented on pmaweb/templates/themes.html:
```diff
@@ -2,75 +2,116 @@
+      <div class="card-header">
+        <h2 class="card-title">Arctic Ocean 2.11a</h2>
+      </div>
+      <div class="card-body">
+        <div class="card-info">
+          <p>Released on 2008-03-03</p>
+          <p>Compatible with phpMyAdmin <strong>2.9 - 2.11</strong>.</p>
+          <p>Author: Michael Keck</p>
+          <p>New theme with new database icons.</p>
+        </div>
+        <div class="card-thumbnail">
+          <a href="/static/images/themes/arctic_ocean.png" class="colorbox cboxElement" title="arctic_ocean theme">
+            <img src="/static/images/themes/arctic_ocean.png" alt="Arctic Ocean thumbnail" class="themeimg">
+          </a>
+        </div>
+      </div>
+      <table class="table">
+        <thead>
+          <tr>
+            <th>File</th>
+            <th class="td-size">Size</th>
+            <th>Verification</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr>
+            <td>
+              <a href="https://files.phpmyadmin.net/themes/arctic_ocean/2.11a/arctic_ocean-2.11a.zip"
+                class="download_popup" data-sha256="ab4611fc954779c4b0271c6bec4378bd6ecb560fc2358b7bd2ddff9fd0e64a9d"
+                data-pgp="">arctic_ocean-2.11a.zip</a></td>
+            <td class="td-size">176.0&nbsp;KB</td>
+            <td>
+              [<a href="https://files.phpmyadmin.net/themes/arctic_ocean/2.11a/arctic_ocean-2.11a.zip.sha256">SHA256</a>]
+            </td>
+          </tr>
+        </tbody>
+      </table>      
+    </div>
+  </div>
 
```
This example can be removed ?

AdrianoCahete replied:
Yep! Removed.

AdrianoCahete committed:
Remove leftover for themes

williamdes commented on pmaweb/templates/components/_dlbox.html:
```diff
@@ -1,3 +1,4 @@
+<!-- TODO: May be removed after the redesign since it's part of header/navbar -->
```
TODO ?

AdrianoCahete replied:
I left because I don't know if you want me to remove or let in there. I didn't find any other place that uses this component.

williamdes commented on pmaweb/templates/components/_icon-library.html:
```diff
@@ -0,0 +1,989 @@
+<svg width="0" height="0" style="display:none;">
```
should it be a .svg ?

AdrianoCahete replied:
Yes. It's an SVG library (or SVG sprite), that I just call each "layer" when I need it. 
It's more resource-saving because we just have one file instead of a lot of SVG's across the project. It's easier to maintain too.

Since it's an inline SVG, we don't need to pay attention to cross-origin protection too.

You can read more [here](https://www.creativebloq.com/features/the-complete-guide-to-svg/6).

williamdes commented on pmaweb/templates/downloads.html:
```diff
@@ -2,203 +2,220 @@
+
+  <h2 id="features">Features</h2>
+  <p>
+    All the features can be found on <a href="https://docs.phpmyadmin.net/en/latest/intro.html?#supported-features">Supported Features</a> page.
```
```suggestion
    All the features can be found on <a href="https://docs.phpmyadmin.net/en/latest/intro.html#supported-features">Supported Features</a> page.
```

williamdes commented on pmaweb/templates/index.html:
```diff
@@ -2,142 +2,147 @@
+              {{ item.date|date:"mm" }}
+            </div>
+            <div class="news-year">
+              <!-- TODO: Needs a if(item.date|date:'yyyy' == current_year) {display:none;} -->
```
todo ?

AdrianoCahete replied:
I left this as TODO because I don't know how to do this on Django (and that news block isn't working on my local machine).

The idea is to hide the year if it's the current one. I may leave it here so you can fix it later.

williamdes reviewed (approved):
I added more comments, but all the diffs LGTM
I think you could add this repository as a remote and rebase onto latest commit to fix the conflicts

AdrianoCahete committed:
Update pmaweb/templates/downloads.html

Co-authored-by: William Desportes <williamdes@wdes.fr>

AdrianoCahete commented:
To be honest, I always have problems with the rebase, :(
I don't know if I want to proceed or prefer that you resolve this.... hahaha

williamdes commented:
> To be honest, I always have problems with the rebase, :(
> I don't know if I want to proceed or prefer that you resolve this.... hahaha

We can rebase or merge locally with resolving conflicts, maybe another or 2 review would be okay so I can proceed with a merge.
Maybe @ibennetch would be able to fix https://github.com/phpmyadmin/website/pull/137#discussion_r530316003 ?
I do not charm pythons :snake: sorry ^^

AdrianoCahete commented:
First, Happy New Year! :)

Now comes the question: is something missing to proceed to merge/deploy @ibennetch @mauriciofauth?

MauricioFauth committed:
Fix broken template include paths

Signed-off-by: Maurício Meneghini Fauth <mauricio@fauth.dev>

mauriciofauth commented:
Hi @AdrianoCahete, I rebased your pull request and I also fixed an issue with the template include paths.

mauriciofauth commented on data/menu.py:
```diff
@@ -23,14 +23,10 @@
 MENU = [
-    ('', 'Home'),
     ('news', 'News'),
-    ('security/', 'Security'),
```
I think we should keep the Security link at the main menu.

AdrianoCahete replied:
My first idea was to move "Security" to the "Support" submenu, but when I was implementing I saw that it will be adding more work than I was planning and move out for this. We can try to go back to this idea again.

Stated [here](https://github.com/phpmyadmin/website/issues/92#issuecomment-713899056):
> Moved "Security" to the "Support" submenu. "Security" as a (main) menu gives the idea that the product has a lot of Security Flaws. If we have it on "Support", gives the idea for the people who search for some support that we have attention to security, but for the main public doesn't need to have that attention, because it's the default behavior (to design a safe system).

mauriciofauth commented on pmaweb/templates/base.html:
```diff
@@ -30,135 +40,158 @@
-        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
+    <nav class="navbar">
+      <div class="nav-container">
+        <button id="menu-handle" class="menuHandle navbar-toggler collapsed" type="button" data-toggle="collapse"
+          data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false"
+          aria-label="Toggle navigation" title="Menu">
           <span class="fa fa-bars"></span>
         </button>
```
I probably broke this when rebasing.

mauriciofauth commented on pmaweb/templates/components/_sponsors.html:
```diff
@@ -0,0 +1,46 @@
+
+    <div class="sponsors-gold">
+      <h3>Gold sponsors</h3>
+      <div class="sponsor-list">
+        <a href="http://www.trialpro.com/" class="link-external"><img src="{% static 'images/trialpro.png' %}" alt="TrialPro" /></a>
+        <a href="https://www.vapehuset.se/" class="link-external"><img src="{% static 'images/vapehuset.png' %}" alt="Vapehuset" /></a>
+      </div>
+
+      <div>
+        <a href="https://www.vapehuset.se/"><img src="{% static 'images/vapehuset.png' %}" alt="Vapehuset" /></a>
+      </div>
+
+      <div>
+        <a href="https://www.4kdownload.com/"><img src="{% static 'images/4kdownload.png' %}" alt="4K Download" /></a>
+      </div>
+
+      <div>
+        <a href="https://goread.io/"><img src="{% static 'images/goread.png' %}" alt="goread" /></a>
+      </div>
```
That is broken as well after the rebase.

mauriciofauth commented on pmaweb/templates/index.html:
```diff
@@ -2,142 +2,147 @@
+            <div class="news-day">
+              {{ item.date|date:"d" }}
+            </div>
+            <div class="news-month" title="{{ item.date|date:'mmmm' }}">
+              {{ item.date|date:"mm" }}
```
These date formats are wrong, see: https://docs.djangoproject.com/en/1.8/ref/templates/builtins/#date

mauriciofauth commented on pmaweb/templates/index.html:
```diff
@@ -2,142 +2,147 @@
+            <div class="news-month" title="{{ item.date|date:'mmmm' }}">
+              {{ item.date|date:"mm" }}
+            </div>
+            <div class="news-year">
+              <!-- TODO: Needs a if(item.date|date:'yyyy' == current_year) {display:none;} -->
+              {{ item.date|date:"yyyy" }}
+            </div>
```
```suggestion
            {% if item.date|date:"Y" != current_year %}
              <div class="news-year">
                {{ item.date|date:"Y" }}
              </div>
            {% endif %}
```

mauriciofauth commented on pmaweb/templates/donate.html:
```diff
@@ -3,13 +3,13 @@
 {% block content %}
 
-<div class="row">
+<div class="content">
 
 <div class="col-md-6">
```
The layout here no longer has two columns. :disappointed:

mauriciofauth commented:
> I may have broken the Announcement list accordion on the Security page (Support > Security) (need to view on a more stable env)

Yes, it is broken.

> Some screenshots doesn't have data because I don't have that data on my local env. I don't know why.

Maybe you forgot to load the test data:
```
./manage.py migrate
./manage.py loaddata pmaweb/fixtures/test_data.json
```
See: https://github.com/phpmyadmin/website#development

mauriciofauth commented:
I think you should add a section on the home page for the Software Freedom Conservancy logo.

AdrianoCahete committed:
Update pmaweb/templates/index.html

Co-authored-by: Maurício Meneghini Fauth <mauricio@fauth.dev>

AdrianoCahete commented:
> 
> 
> > I may have broken the Announcement list accordion on the Security page (Support > Security) (need to view on a more stable env)
> 
> Yes, it is broken.
> 
> > Some screenshots doesn't have data because I don't have that data on my local env. I don't know why.
> 
> Maybe you forgot to load the test data:
> 
> ```
> ./manage.py migrate
> ./manage.py loaddata pmaweb/fixtures/test_data.json
> ```
> 
> See: https://github.com/phpmyadmin/website#development

I tried, but it cannot run on Windows (`ModuleNotFoundError: No module named 'urllib2'`). But, to be honest, I'm receiving an error now when I try to run the website on my machine (not happened before or I managed to fix it on my end).

I'm a little busy right now, but I'll try to fix that in the next weeks.

AdrianoCahete commented:
> I think you should add a section on the home page for the Software Freedom Conservancy logo.

Right aligned on the same line/row as the Sponsors, maybe?

williamdes commented:
> I tried, but it cannot run on Windows (ModuleNotFoundError: No module named 'urllib2'). But, to be honest, I'm receiving an error now when I try to run the website on my machine (not happened before or I managed to fix it on my end).

Could be a python3 issue instead you would probably need to use python2

mauriciofauth commented:
> > I think you should add a section on the home page for the Software Freedom Conservancy logo.
> 
> Right aligned on the same line/row as the Sponsors, maybe?
> 
> 

I thought about a full width banner with the logo and some text.

williamdes commented:
Hi @AdrianoCahete 
I hope you are doing good nowdays ! :)

This pr is outdated by #186 that did improve the website.
That said, you can re-open a PR and pick some of your wording changes.
But I am not sure if we want to change the UI of the website.
The logo would be nice to update if everyone likes it

AdrianoCahete commented:
Hey, congrats for the update! I'll try to see something in the next weeks, but can't promise :(
